ts/output/common/FontData.ts

```diff
@@ -561,12 +621,50 @@ export class FontData<C extends CharOptions, V extends VariantData<C>, D extends
    */
   public static charOptions(font: CharMap<CharOptions>, n: number): CharOptions {
     const char = font[n];
+    if (!Array.isArray(char)) {
+      throw Error(`Character data hasn't been loaded for 0x${n.toString(16).toUpperCase()}`);
+    }
     if (char.length === 3) {
       (char as any)[3] = {};
     }
     return char[3];
   }
 
+  /**
+   * Define the dynamic file information.
+   *
+   * @param {DynamicFileDef[]} dynamicFiles   The definitions to make
+   * @return {DynamicFileList}                The object of dynamic file data
+   */
+  public static defineDynamicFiles(dynamicFiles: DynamicFileDef[]): DynamicFileList {
+    const list: DynamicFileList = {};
+    dynamicFiles.map(([file, variants, delimiters]) => {
+      list[file] = {
+        file, variants, delimiters: delimiters || [],
+        promise: null, failed: false, setup: ((_font) => { list[file].failed = true; })
+      } as DynamicFile;
+    });
+    return list;
+  }
+
+  /**
+   * Create the setup function for a given dynamically loaded file
+   *
+   * @param {string} file           The file being loaded
+   * @param {CharMapMap} variants   The character data to be added
+   *
+   * @template C  The CharOptions type
+   * @template D  The DelimiterData type
+   */
+  public static dynamicSetup<C extends CharOptions, D extends DelimiterData>(
+    file: string, variants: CharMapMap<C>, delimiters: DelimiterMap<D> = []
+  ) {
+    this.dynamicFiles[file].setup = (font) => {
+      Object.keys(variants).forEach(name => font.defineChars(name, variants[name]));
```

Comment:
Can we start using `Object.entries` in situations like this? Or we should just have Maps everywhere.

Reply:
We need to change the tsconfig to include additional libraries in the `libs` array.  I used `es2019` in the font-tools repository in order to allow for this, and would love to have it in MathJax-src as well.  Shall we do that?

Reply:
Here, the `Object.keys()` are for the `variants` object, which is a `CharMapMap`, and so that would need to be changes to a `Map()` as well.  That will affect all the font definitions, and again, I'd like to put this off until all the font PRs are merged, do a pass over the code for these kinds of changes, and then rebuild the font packages only once in response.

Reply:
Yes, let's add `es2019`.